src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
actually no, i think this is better, since these functions take `extension` as an argument.